doc/source/changes/version_0_21.rst.inc

```diff
@@ -0,0 +1,116 @@
+    >>> arr.set_axes(a=row, b=column)
+    row\\column | c0 | c1 | c2
+            r0 |  0 |  1 |  2
+            r1 |  3 |  4 |  5
+    >>> arr.set_axes([(x.a, row), (x.b, column)])
+    row\\column | c0 | c1 | c2
+            r0 |  0 |  1 |  2
+            r1 |  3 |  4 |  5
+    >>> arr.set_axes({x.a: row, x.b: column})
+    row\\column | c0 | c1 | c2
+            r0 |  0 |  1 |  2
+            r1 |  3 |  4 |  5
+    Replace all axes (list of axes or AxisCollection)
+    >>> arr.set_axes([row, column])
+    row\\column | c0 | c1 | c2
+            r0 |  0 |  1 |  2
+            r1 |  3 |  4 |  5
+    >>> arr2 = ndrange([row, column])
+    >>> arr.set_axes(arr2.axes)
+    row\\column | c0 | c1 | c2
+            r0 |  0 |  1 |  2
+            r1 |  3 |  4 |  5
+
+* implemented from_string() method to create an array from a string:
+
+    >>> from_string('''age,nat\\sex, M, F
+    ...                0,  BE,       0, 1
+    ...                0,  FO,       2, 3
+    ...                1,  BE,       4, 5
+    ...                1,  FO,       6, 7''')
+    age | nat\sex | M | F
+      0 |      BE | 0 | 1
+      0 |      FO | 2 | 3
+      1 |      BE | 4 | 5
+      1 |      FO | 6 | 7
+
+* allowed to use a regular expression in split_axis method (closes :issue:`106`):
+
+    >>> combined = ndrange('a_b = a0b0..a1b2')
+    >>> combined
+        a_b | a0b0 | a0b1 | a0b2 | a1b0 | a1b1 | a1b2
+            |    0 |    1 |    2 |    3 |    4 |    5
+    >>> combined.split_axis(x.a_b, regex='(\w{2})(\w{2})')
+        a\\b | b0 | b1 | b2
+         a0 |  0 |  1 |  2
+         a1 |  3 |  4 |
+
+* implemented Axis.by() which is equivalent to axis[:].by()
+
+    >>> Axis('age', range(10)).by(3, 4)
+    (age.i[0:3], age.i[4:7], age.i[8:10])
+
+Miscellaneous improvements
+--------------------------
+
+* added installation instructions (closes :issue:`101`).
+
+* viewer : make shortcuts work even when the focus is not on the array editor widget
+  (ie it is on the array list, or on the interactive console) (closes :issue:`102`).
+
+* allowed matrix multiplication (@ operator) between arrays with dimension != 2 (closes :issue:`122`).
+
+* viewer : automatically display plots done in the viewer console in a separate window
+  (unless "%matplotlib inline" is used). Plots can be done via the array widget
+  (using shortcut CTRL+P or right click) or from the console:
+
+    >>> arr = ndtest((3, 3))
+    >>> arr.plot()
+
+  Now both methods generate a plot in separate window.
+
+* improved LArray.plot to get nicer plots by default.
+  The axes are transposed compared to what they used to, because the last axis is often used for time series.
+  Also it considers a 1D array like a single series, not N series of 1 point.
+
+* added possibility to tag multiple groups with an axis in one shot (backward incompatible)
+
+    >>> year  = Axis('year', '2001 .. 2010')
+    >>> year.group('2001,2002;2003:2008;2009,2010')
```

Comment:
I think we should deprecate .group (see #148), and the commit actually included two distinct changes:
* interpret string syntax in axis[a_string], which means you can do:
```python
>>> year = Axis('year', '2001..2010')
>>> year['2001:2004']
year[2001:2004]
```
* change the axis of multiple groups via a single axis[groups]

```python
>>> groups = year[2001:2004], year[2008,2009]
>>> groups
(year[2001:2004], year[2008, 2009])
>>> x.time[groups]
(x.time[2001:2004], x.time[2008, 2009])
```
However (and this is a bit silly that I did not make it work -- but to my defense, I did not expect you to merge that PR), what you used as an example is broken at this point, because the result is not a valid index in an array:
```python
>>> year['2001:2004;2008;2009']
year[slice(2001, 2004, None), [2008, 2009]]
>>> arr = ndrange(year)
>>> arr.sum(year['2001:2004;2008,2009'])
TypeError                                 Traceback (most recent call last)
...
TypeError: unhashable type: 'slice'
>>> # this works though
... arr.sum(tuple(year['2001:2004;2008,2009']))
year | 2001:2004 | 2008,2009
     |         6 |        15
```